### PR TITLE
(WIP) docs: improve docs for `riot-rs-threads`

### PR DESCRIPTION
**Pull request by elenaf9**
Improve API documentation for `riot-rs-threads` and `riot-rs-runqueue`. 
I'll work on next to extract the essence about scheduling in riot-rs into a markdown file in the book (most likely as _User Guide_ and _Developer Guide_?) but I thought that it might make sense to already open this as a draft PR now.

Open Questions:

- Should the user/developer guide be in the `book/` or `docs/` folder?
- I don't fully understand `Threads::flag_wait_one` - @kaspar030 could you help me here?

### CODE REVIEW

**Comment by kaspar030** [comment]
> * I don't fully understand `Threads::flag_wait_one` - @kaspar030 could you help me here?

`Threads::flag_wait_any(mask)` returns when any of `mask` are set, and returns all of those (`set & mask`). `Threads::flag_wait_one(mask)` returns when any of `mask` are set, and returns exactly one of those (`ffs (set & mask)`), leaving the not returned active.

This is (maybe) useful if the handler would rather handle the flags one-by-one.

**Comment by kaspar030** [comment]
> * Should the user/developer guide be in the `book/` or `docs/` folder?

Let's have them in the book. That's markdown after all. There are sections for user/developer docs already.

File: src/riot-rs-rt/src/threading.rs [hunk 1]
```diff
@@ -3,6 +3,10 @@ use riot_rs_threads::{start_threading, THREAD_FNS};
 /// # Safety
 ///
 /// The caller must ensure that this function is only called once.
+///
+/// # Panics
+///
+/// Panics if no thread exists.
```
Comment (kaspar030) [inline src/riot-rs-rt/src/threading.rs:9]: This is a bug btw. :)

**Comment by kaspar030** [comment]
Maybe it makes sense to split this PR into smaller chunks - maybe one per crate and one for the book?

File: src/riot-rs-rt/src/threading.rs (hunk 1 above)
Comment (elenaf9) [reply src/riot-rs-rt/src/threading.rs:9]: Extracted into #148 :)

**Comment by elenaf9** [comment]
> Maybe it makes sense to split this PR into smaller chunks - maybe one per crate and one for the book?

Good idea.

Closing this PR in favor of #149, #150 and #152

**Pull request closed without merging**